gitlab/v4/objects/job_token_scope.py

```diff
@@ -27,3 +35,23 @@ class ProjectJobTokenScopeManager(GetWithoutIdMixin, UpdateMixin, RESTManager):
 
     def get(self, **kwargs: Any) -> ProjectJobTokenScope:
         return cast(ProjectJobTokenScope, super().get(**kwargs))
+
+
+class AllowlistedProject(ObjectDeleteMixin, RESTObject):
+    _id_attr = "target_project_id"  # note: only true for create endpoint
+
+    def get_id(self) -> int:
+        """Returns the id of the resource. This override deals with
+        the fact that either an `id` or a `target_project_id` attribute
+        is returned by the server depending on the endpoint called."""
+        try:
```

Comment:
Unsure if should use the  `get_id()` method in the base class?

Maybe something like:

```
  id = cast(int, super().get_id())
  if id is not None:
    return None
  return cast(int, self.id)
```

But I'm okay without. Just thinking aloud. Thoughts?

Reply:
I'm in favour of re-using what we have, and I've got unit-tests for this action so I'll have a play 👍